pkg/admiralctl/migrate/migrate.go

```diff
@@ -0,0 +1,694 @@
+			return err
+		}
+		if job.Spec.ManualSelector == nil || !*job.Spec.ManualSelector {
+			if err = removeGenerateSelectorOfJob(workload); err != nil {
+				return err
+			}
+		}
+	}
+
+	if workload.GetKind() == common.ServiceAccountKind {
+		secrets, exist, _ := unstructured.NestedSlice(workload.Object, "secrets")
+		// If 'secrets' exists in ServiceAccount, remove the automatic generation secrets(e.g. default-token-xxx)
+		if exist && len(secrets) > 0 {
+			tokenPrefix := fmt.Sprintf("%s-token-", workload.GetName())
+			for idx := 0; idx < len(secrets); idx++ {
+				if strings.HasPrefix(secrets[idx].(map[string]interface{})["name"].(string), tokenPrefix) {
+					secrets = append(secrets[:idx], secrets[idx+1:]...)
+				}
+			}
+			_ = unstructured.SetNestedSlice(workload.Object, secrets, "secrets")
+		}
+	}
+
+	for i := range extraHooks {
+		extraHooks[i](workload)
+	}
+	return nil
+}
+
+// getLabelValue retrieves the value via 'labelKey' if exist, otherwise returns an empty string.
+func getLabelValue(labels map[string]string, labelKey string) string {
+	if labels == nil {
+		return ""
+	}
+
+	return labels[labelKey]
+}
+
+func removeGenerateSelectorOfJob(workload *unstructured.Unstructured) error {
+	matchLabels, exist, err := unstructured.NestedStringMap(workload.Object, "spec", "selector", "matchLabels")
+	if err != nil {
+		return err
+	}
+	if exist {
+		if getLabelValue(matchLabels, "controller-uid") != "" {
+			delete(matchLabels, "controller-uid")
+		}
+		err = unstructured.SetNestedStringMap(workload.Object, matchLabels, "spec", "selector", "matchLabels")
+		if err != nil {
+			return err
+		}
+	}
+
+	templateLabels, exist, err := unstructured.NestedStringMap(workload.Object, "spec", "template", "metadata", "labels")
+	if err != nil {
+		return err
+	}
+	if exist {
+		if getLabelValue(templateLabels, "controller-uid") != "" {
+			delete(templateLabels, "controller-uid")
+		}
+
+		if getLabelValue(templateLabels, "job-name") != "" {
+			delete(templateLabels, "job-name")
+		}
+
+		err = unstructured.SetNestedStringMap(workload.Object, templateLabels, "spec", "template", "metadata", "labels")
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// convertToTypedObject converts an unstructured object to typed.
+func convertToTypedObject(in, out interface{}) error {
+	if in == nil || out == nil {
+		return fmt.Errorf("convert objects should not be nil")
+	}
+
+	switch v := in.(type) {
+	case *unstructured.Unstructured:
+		return runtime.DefaultUnstructuredConverter.FromUnstructured(v.UnstructuredContent(), out)
+	case map[string]interface{}:
+		return runtime.DefaultUnstructuredConverter.FromUnstructured(v, out)
+	default:
+		return fmt.Errorf("convert object must be pointer of unstructured or map[string]interface{}")
+	}
+}
+
+// buildPropagationPolicy build PropagationPolicy according to resource and cluster
+func buildPropagationPolicy(policyName, namespace, cluster, schedulingMode string) *fedcorev1a1.PropagationPolicy {
```

Comment:
You can try to merge the `buildPropagationPolicy` and `buildClusterPropagationPolicy` into a common function.